pkg/kepval/prrs/validations/yaml.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validations
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"k8s.io/enhancements/pkg/kepval/util"
+)
+
+var (
+	mandatoryKeys = []string{"kep-number"}
+)
+
+func ValidateStructure(parsed map[interface{}]interface{}) error {
+	for _, key := range mandatoryKeys {
+		if _, found := parsed[key]; !found {
+			return util.NewKeyMustBeSpecified(key)
+		}
+	}
+
+	for key, value := range parsed {
+		// First off the key has to be a string. fact.
+		k, ok := key.(string)
+		if !ok {
+			return util.NewKeyMustBeString(k)
+		}
+
+		// figure out the types
+		switch strings.ToLower(k) {
+		case "alpha", "beta", "stable":
```

Comment:
I think the validation should be:

- Find the `stage` from kep.yaml
- Check for presence and validity of PRR approver for that stage, if `latest-milestone` is > 1.20

Reply:
That makes sense to me, but I think this logic doesn't belong here.

This part is more of a syntax validation.
The actual semantic validation depending on the kep.yaml is happening in the test itself - let me add that logic there.

Reply:
Added in the test itself (main_test.go)